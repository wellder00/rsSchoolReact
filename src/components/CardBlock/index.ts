export { default as CardBlock } from './CardBlock';
