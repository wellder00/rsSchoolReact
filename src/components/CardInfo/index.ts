export { default as CardInfo } from './CardInfo';
