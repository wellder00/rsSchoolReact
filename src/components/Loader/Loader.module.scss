.loadingIcon {
  font-size: 30px;
  padding: 20px 0;
}

.loader {
  width: 100px;
  height: 100px;
  border: 10px solid #3498db;
  border-top: 4px solid transparent;
  border-radius: 50%;
  animation: spin 2s linear infinite;
}

.loader-circle {
  width: 50px;
  height: 50px;
  background-color: #3498db;
  border-radius: 50%;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
