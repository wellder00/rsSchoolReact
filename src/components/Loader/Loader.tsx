import styles from './Loader.module.scss';

const Loader = () => {
  return (
    <div className={styles.loading}>
      <div className={styles.loadingIcon}>Loading...</div>
      <div className={styles.loader}>
        <div className={styles.loaderCircle}></div>
      </div>
    </div>
  );
};

export default Loader;
