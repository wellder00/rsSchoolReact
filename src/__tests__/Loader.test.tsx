import { render, screen } from '@testing-library/react';
import { Loader } from '../components/Loader';
import '@testing-library/jest-dom';

test('should show loading indicator', () => {
  const { asFragment } = render(<Loader />);
  expect(asFragment()).toMatchSnapshot();
  const loadingElement = screen.getByTestId('loading-icon');
  expect(loadingElement).toBeInTheDocument();
});
