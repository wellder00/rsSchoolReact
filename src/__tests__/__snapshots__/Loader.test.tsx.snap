// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show loading indicator 1`] = `
<DocumentFragment>
  <div
    class="loading"
  >
    <div
      class="loadingIcon"
      data-testid="loading-icon"
    >
      Loading...
    </div>
    <div
      class="loader"
    >
      <div
        class="loaderCircle"
      />
    </div>
  </div>
</DocumentFragment>
`;
